### PR TITLE
#16 & #17 resolved

### LINKED ISSUE #16: Add new classes to datalist.
# Expected functionality

When a user types in a class name and then clicks the "Insert" button, the class name should be added to the `<datalist>` element in `index.html` as an option just like the others (Math, Science, etc.).

# Example

So if I type in Physics like this:

image

Then press Insert:

image

Physics should be added to the dropdown list:

image

I suggest working with `script.js` for this functionality with the Insert button event handler.

### LINKED ISSUE #17: Alignment abstraction.
Remove the `align="center"` attribute from the the title and table in `index.html` and instead align the title and table in the center from the CSS file.

### PR DESCRIPTION
**Pull request by ishaanbedi**
Changed the alignment from CSS file and made the class name embed in the dataset

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -23,14 +23,18 @@ <h1 align="center">Mega Todo</h1>
         </tr>
         <tr id="baseRow">
             <td id="assignmentCell"><input class="assignmentName" type="text" /></td> <!-- Input -->
-            <td> 
-                <input list="Classes" type="text" name="class section"  class="className">
-                <datalist id="Classes">
+
+            <td>
+                <input list="citynames" name="cityname" id="cityname" class="className">
+                <datalist id="citynames" class="DataList">
```
Comment (rzmk) [inline index.html]: Semantically better. Commit this along with the below Javascript commit. 
```suggestion
                <input list="classes" class="className">
                <datalist id="classes" class="DataList">
```

File: js/script.js [hunk 2]
```diff
@@ -23,6 +23,15 @@ buttonInsert.addEventListener('click', function () {
         cell5.innerText = document.querySelector('.priorityInput').value;
         cell6.innerText = document.querySelector('.estimatedTimeInput').value;
         cell7.innerText = document.querySelector('.status').value;
+
+
+        var ele = document.createElement("OPTION");
+        ele.setAttribute("value", document.querySelector('.className').value);
+        var par = document.createTextNode(document.querySelector('.className').value);
+        ele.appendChild(par);
+        document.querySelector("#citynames").appendChild(ele);
```
Comment (rzmk) [inline js/script.js]: Commit this along with the above in `index.html`.
```suggestion
        document.querySelector("#classes").appendChild(ele);
```

**Review by rzmk — CHANGES_REQUESTED** [review]
Great work! I've added some changes to make things cleaner and easier to understand, so commit these and this should be good to go! 🚀

**Comment by ishaanbedi** [comment]
Hey! I've edited the required code.

**Comment by rzmk** [comment]
@ishaanbedi Looks great!

One issue I find is that if you select a class and insert, it'll add another copy of it. Would you like to try checking against the classes on insert and make the class addition conditional?

Example:

image

**Comment by ishaanbedi** [comment]
Alright! Working on it

**Comment by ishaanbedi** [comment]
Hi! I've fixed it!

File: js/script.js [hunk 3]
```diff
@@ -23,6 +29,24 @@ buttonInsert.addEventListener('click', function () {
...
+
+        for (let i = 0; i < dataSetArray.length; i++) {
+            if (String(dataSetArray[i]) === String(valueOfClassName)) {
+                flag = 1;
+            }
+
+        }
+        if (flag === 0) {
+            dataSetArray.push(valueOfClassName);
+            var ele = document.createElement("OPTION"); var ele = document.createElement("OPTION");
```
Comment (rzmk) [inline js/script.js]: Curious why there's a declaration of `ele` twice.

**Review by rzmk — CHANGES_REQUESTED** [review]

File: js/script.js (hunk 3 above)
Comment (ishaanbedi) [reply js/script.js]: Might have been by mistake! 
Corrected it

**Review by rzmk — APPROVED** [review]
Good to go! 🚀
Great work once again! 😁

**Pull request merged**